dev-tools/mage/manifest/manifestspecs.go

```diff
@@ -0,0 +1,163 @@
+package manifest
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/magefile/mage/mg"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+type Dependency struct {
+	Prefix   string `json:"prefix"`
+	BuildUri string `json:"build_uri"`
+}
+
+type Project struct {
+	Branch                       string             `json:"branch"`
+	CommitHash                   string             `json:"commit_hash"`
+	CommitUrl                    string             `json:"commit_url"`
+	ExternalArtifactsManifestUrl string             `json:"external_artifacts_manifest_url"`
+	BuildDurationSeconds         int                `json:"build_duration_seconds"`
+	Packages                     map[string]Package `json:"packages"`
+	Dependencies                 []Dependency       `json:"dependencies"`
+}
+
+type Package struct {
+	Url          string   `json:"url"`
+	ShaUrl       string   `json:"sha_url"`
+	AscUrl       string   `json:"asc_url"`
+	Type         string   `json:"type"`
+	Architecture string   `json:"architecture"`
+	Os           []string `json:"os,omitempty"`
+	Classifier   string   `json:"classifier,omitempty"`
+	Attributes   struct {
+		IncludeInRepo string `json:"include_in_repo,omitempty"`
+		Oss           string `json:"oss,omitempty"`
+		ArtifactNoKpi string `json:"artifactNoKpi,omitempty"`
+		Internal      string `json:"internal,omitempty"`
+		ArtifactId    string `json:"artifact_id,omitempty"`
+		Group         string `json:"group,omitempty"`
+	} `json:"attributes,omitempty"`
+}
+
+type ManifestResponse struct {
+	Branch               string             `json:"branch"`
+	ReleaseBranch        string             `json:"release_branch"`
+	Version              string             `json:"version"`
+	BuildId              string             `json:"build_id"`
+	StartTime            string             `json:"start_time"`
+	EndTime              string             `json:"end_time"`
+	BuildDurationSeconds int                `json:"build_duration_seconds"`
+	ManifestVersion      string             `json:"manifest_version"`
+	Prefix               string             `json:"prefix"`
+	Projects             map[string]Project `json:"projects"`
+}
+
+type PackageComponent struct {
+	PackageName    string `yaml:"package_name"`
+	PackageRepo    string `default:"" yaml:"package_repo"`
+	PackageVersion string `yaml:"package_version"`
+	PackageBuildId string `default:"" yaml:"package_build_id"`
+}
+
+func downloadFile(url string, filepath string) (string, error) {
+	outFile, fileErr := os.Create(filepath)
+	if fileErr != nil {
+		return "", fmt.Errorf("failed to create destination file %w", fileErr)
+	}
+	defer func() {
+		outFile.Close()
+	}()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return filepath, fmt.Errorf("failse to download manifest [%s]\n %w\n", url, err)
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			panic(err)
+		}
+	}(resp.Body)
+
```

Comment:
Maybe we can add a check on the http status code here:

```suggestion
if resp.StatusCode != http.StatusOK {
		return "", fmt.Errorf("bad status code for %q:  %d :%s ", url, resp.StatusCode, resp.Status)
	}
```